# Cryptocurrency Market Anomaly Detection
## First step of the project is data exploration

In [1]:
import pandas as pd
import matplotlib as plt
import yfinance as yf

### To start a project I need to import cryptocurrency data

In [2]:
btc_data = yf.download("BTC-USD", start = "2017-05-01")

[*********************100%***********************]  1 of 1 completed


In [3]:
btc_data.info()

<class 'pandas.DataFrame'>
DatetimeIndex: 3322 entries, 2017-05-01 to 2026-06-04
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   (Close, BTC-USD)   3322 non-null   float64
 1   (High, BTC-USD)    3322 non-null   float64
 2   (Low, BTC-USD)     3322 non-null   float64
 3   (Open, BTC-USD)    3322 non-null   float64
 4   (Volume, BTC-USD)  3322 non-null   int64  
dtypes: float64(4), int64(1)
memory usage: 155.7 KB


In [4]:
btc_data.head()

Price,Close,High,Low,Open,Volume
Ticker,BTC-USD,BTC-USD,BTC-USD,BTC-USD,BTC-USD
Date,,,,,
2017-05-01,1421.599976,1434.319946,1348.300049,1348.300049,713624000
2017-05-02,1452.819946,1473.900024,1415.689941,1421.030029,477337984
2017-05-03,1490.089966,1492.770020,1447.489990,1453.780029,583795968
2017-05-04,1537.670044,1608.910034,1490.719971,1490.719971,933548992
2017-05-05,1555.449951,1618.030029,1530.310059,1540.869995,946035968


In [5]:
btc_data.tail()

Price,Close,High,Low,Open,Volume
Ticker,BTC-USD,BTC-USD,BTC-USD,BTC-USD,BTC-USD
Date,,,,,
2026-05-31,73579.687500,74153.500000,73315.015625,73753.750000,17445435650
2026-06-01,71319.773438,73969.570312,70599.507812,73580.210938,43378836121
2026-06-02,66703.656250,71334.953125,66127.265625,71321.031250,54990073882
2026-06-03,64014.367188,67402.929688,64009.675781,66694.007812,47411556213
2026-06-04,63370.460938,64659.058594,61351.628906,64038.921875,64713494528


In [6]:
btc_data.index

DatetimeIndex(['2017-05-01', '2017-05-02', '2017-05-03', '2017-05-04',
               '2017-05-05', '2017-05-06', '2017-05-07', '2017-05-08',
               '2017-05-09', '2017-05-10',
               ...
               '2026-05-26', '2026-05-27', '2026-05-28', '2026-05-29',
               '2026-05-30', '2026-05-31', '2026-06-01', '2026-06-02',
               '2026-06-03', '2026-06-04'],
              dtype='datetime64[s]', name='Date', length=3322, freq=None)

In [8]:
Last_date = btc_data.index[0] - btc_data.index[-1]
print(Last_date)

-3321 days +00:00:00


#### After importing the BTC data (imported from 01-05-2017 - 04-06-2026 - begin of higher changes in BTC till today), I have checked the data imported and also number of days - to be sure I have imported all necessery data - both shows 3322 